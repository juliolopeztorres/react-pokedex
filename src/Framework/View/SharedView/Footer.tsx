import React from 'react'
import { hot } from "react-hot-loader/root";

const Footer = () => <div className="bottom-0 bg-gray-800 p-4 text-gray-200 text-sm font-mono dark:bg-gray-950">
    <span>Julio + React + Tailwind</span>
  </div>

export default hot(Footer)
