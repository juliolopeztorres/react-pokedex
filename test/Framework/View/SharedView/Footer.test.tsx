import React from 'react'
import { createRoot, Root } from "react-dom/client"
import { act } from "@testing-library/react"

import Footer from "../../../../src/Framework/View/SharedView/Footer"

let root: Root
let container: HTMLElement
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div")
  document.body.appendChild(container)

  root = createRoot(container)
})

afterEach(() => {
  // cleanup on exiting
  act(() => root.unmount())

  container.remove()
})

it('should render', () => {
  act(() => root.render(<Footer/>))

  expect(container.textContent).toContain('Julio + React + Tailwind')
  expect(container.childElementCount).toBe(1)

  expect(container).toMatchSnapshot()
})
