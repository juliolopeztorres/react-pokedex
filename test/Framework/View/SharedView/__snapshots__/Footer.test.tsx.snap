// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    class="bottom-0 bg-gray-800 p-4 text-gray-200 text-sm font-mono dark:bg-gray-950"
  >
    <span>
      Julio + React + Tailwind
    </span>
  </div>
</div>
`
